cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -161,7 +167,7 @@ type_id to_type_id(SchemaElement const& schema,
   switch (physical) {
     case BOOLEAN: return type_id::BOOL8;
     case INT32: return type_id::INT32;
-    case INT64: return type_id::INT64;
+    case INT64: return arrow.value_or(type_id::INT64);
```

Comment:
I think you'll also have to check the switch above. Arrow will almost certainly add the logical type, which will likely be TIME_MILLIS, so we're not likely to reach this statement to do the override.

Reply:
Hi Ed, I checked arrow's codebase and duration type is encoded as int64 type (no logical or converted types involved). I also confirmed this using printfs. 🙂

Reply:
Wow, suprising. Never mind then :grin: